exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/handlers/SqlHandlerConfig.java

```diff
@@ -52,12 +59,81 @@ public QueryContext getContext() {
     return context;
   }
 
+  private RuleSet attemptToGetRules(PlannerPhase phase, Collection<StoragePlugin> plugins) throws PluginException{
```

Comment:
@vvysotskyi so do we keep auto disabling but drop retry here? We've seen JDBC plugins that point to an unresponsive remote DB fail with the first exception coming from rule collection (because, unlike most plugins, JDBC already tries to contact the remote DB for this operation). And of course before DRILL-8234, JDBC plugins not even involved in a query could cause it to fail during rule collection.

Reply:
I meant to drop both retry and auto-disabling logic. I have checked JDBC plugin, and it has several places where the data source is created. One of them is `getOptimizerRules` which is used here, and another one is `registerSchemas`, which is called during validation, before this place. So if we have something broken, it should fail before calling this method.